src/lightning_lite/strategies/tpu_spawn.py

```diff
@@ -0,0 +1,202 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import io
+import os
+from typing import Any, Dict, List, Mapping, Optional, Sequence, Union
+
+import torch
+from torch import Tensor
+from torch.nn import Module
+from torch.utils.data import DataLoader
+
+from lightning_lite.accelerators import Accelerator
+from lightning_lite.plugins.environments import XLAEnvironment
+from lightning_lite.plugins.io.checkpoint_plugin import CheckpointIO
+from lightning_lite.plugins.io.xla_plugin import XLACheckpointIO
+from lightning_lite.plugins.precision import Precision
+from lightning_lite.strategies.ddp_spawn import DDPSpawnStrategy
+from lightning_lite.strategies.launchers.xla import _XLALauncher
+from lightning_lite.strategies.strategy import TBroadcast
+from lightning_lite.utilities import _TPU_AVAILABLE
+from lightning_lite.utilities.apply_func import apply_to_collection
+from lightning_lite.utilities.data import has_len
+from lightning_lite.utilities.distributed import ReduceOp
+from lightning_lite.utilities.rank_zero import rank_zero_only
+from lightning_lite.utilities.types import _PATH
+
+if _TPU_AVAILABLE:
+    import torch_xla.core.xla_env_vars as xenv
+    import torch_xla.core.xla_model as xm
+    from torch_xla.core.xla_model import rendezvous
+    from torch_xla.distributed.parallel_loader import MpDeviceLoader
+else:
+    xm, xmp, MpDeviceLoader, rendezvous = [None] * 4
+
+
+class TPUSpawnStrategy(DDPSpawnStrategy):
```

Comment:
Let's consider naming this just TPUStrategy?

In the future, it will not subclass DDPSpawn strategy, and it has it's own launcher implementation anyway. Plus, the implementation differs quite significantly from DDPSpawnStrategy

Reply:
I agree, but would prefer XLAStrategy. Having TPU in the name is confusing since TPU is an accelerator, not a strategy